MoreLinq.Test/SliceTest.cs

```diff
@@ -125,5 +125,23 @@ public void TestSliceListOptimization()
             Assert.AreEqual(sliceCount, result.Count());
             Assert.IsTrue(result.SequenceEqual(Enumerable.Range(5, sliceCount)));
         }
+
+#if IREADONLY
+        /// <summary>
+        /// Verify that slice is optimized for <see cref="IReadOnlyList{T}"/> implementations and does not
+        /// unnecessarily traverse items outside of the slice region.
+        /// </summary>
+        [Test]
+        public void TestSliceReadOnlyListOptimization()
```

Comment:
Can we avoid copy-paste duplication between this and [`TestSliceListOptimization`](https://github.com/Arithmomaniac/MoreLINQ/blob/cf6dfa5c70020f9cbabba29ca1cd087535df1e46/MoreLinq.Test/SliceTest.cs#L118)?

Reply:
Sure. Would you prefer using `TestCaseData` or a private static `TestSliceOptimization_Impl` method?

Reply:
I imagine `TestCaseData` will be simpler for now.